Documentation for using managed identity (#61)
UKHO/EventHub-Logging-Provider: README.md

The Event Hub Log Provider provides a logging sink for the Microsoft.Extensions.Logging.Abstractions. Logs are sent to Event Hub as a JSON message. The EventHubLogProvider provides a number of standard properties to enrich every log message, and provides a mechanism to add application specific custom properties to logs.

> The Event Hub Log Provider supports Managed Identity and connection string/SAS URI authentication for Event Hub and Azure Storage container access. However, it is required to use either Managed Identity for both Event Hub and Azure Storage or neither.

Reviewer: ```suggestion
The Event Hub Log Provider supports these authentication methods

- Managed Identity (Both Event Hub & Storage Account)
- Connection string for Event Hub and SAS URI authentication for Azure Storage container access.

Authentication methods cannot be mixed between Event Hub and Storage Account, either both use Managed Identity or they use the other authentification method. 
```

> Event hub log provider now supports authentication using both Managed Service Identity or connection string/SAS Uri to Event hub or Azure storage container respectively. However they both should use either Managed Identity to both Eventhub and Azure storage or neither. 

Reviewer 2: ```suggestion
Event Hub Log Provider now supports authentication using both Managed Service Identity or connection string/SAS Uri to Event Hub or Azure storage container respectively. However they both should use either Managed Identity to both Event Hub and Azure storage or neither. 
```

> Event Hub Log Provider now supports authentication using both Managed Service Identity or connection string/SAS Uri to Event Hub or Azure storage container respectively. However they both should use either Managed Identity to both Event Hub and Azure storage or neither. 

Reviewer 3: ```suggestion
The Event Hub Log Provider supports Managed Identity and connection string/SAS URI authentication for Event Hub and Azure Storage container access. However, it is required to use either Managed Identity for both Event Hub and Azure Storage or neither.
```

> Configuring using managed identity or connection string is shown below.

Reviewer 3: ```suggestion
Instructions to configure the services with either a managed identity or a connection string are below.
```

The Azure Storage Logging Provider is a storage logging provider that stores messages with size equal or greater than 1Mb into an Azure storage container. Finally, it updates the log entry with the azure storage blob details. Enabling the Azure storage provider is optional.

The provider can be enabled by providing the AzureStorageLogProviderOptions.

> It supports both managed identity or SAS url for authentication. Below are example for each.

Reviewer 3: Already mentioned at the top of the readme, do we need to repeat it here?

Author: Removed and tweaked the text accordingly

> It supports bothe managed identity or SAS url for authentication. Below are example for each.

Reviewer 2: ```suggestion
It supports both managed identity or SAS url for authentication. Below are example for each.
```

  }
```

### Configuration using Managed identity

 The AzureStorageLogProviderOptions model consists of:

 ```
options.AzureStorageLogProviderOptions = new AzureStorageLogProviderOptions(
                        // Azure storage blob container uri
                        new Uri(builder.Configuration.GetValue<String>("AzureStorageOptions:BlobContainerUri"),
                        // The Azure managed identity credential to use for authorization.
                        new DefaultAzureCredential(),
                        //Enables (true) or disables(false) the Azure storage logging provider
                        builder.Configuration.GetValue<String>("AzureStorageOptions:AzureStorageLoggerEnabled"),
                        //A template for the messages that are successfully stored
                        builder.Configuration.GetValue<String>("AzureStorageOptions:SuccessMessageTemaplate"),
                        //A template for the messages that failed to be stored*                      
                        builder.Configuration.GetValue<String>("AzureStorageOptions:FailedMessageTemplate")
                        );

```

> ### Example of a configuration section(json) 

Reviewer 3: Should this have 4 # as per Ln. 323?

```suggestion
#### Example of a configuration section(json) 
```

```json
    "AzureStorageOptions": {
    "BlobContainerUri": "blob container uri", //This is likely to be similar to "https://{account_name}.blob.core.windows.net/{container_name}"
    "Enabled": true,
    "SuccessfulMessageTemplate": "Azure Storage Logging: A blob with the error details was created at {{BlobFullName}}. Reason: ErrorMessageEqualOrGreaterTo1MB ResponseMessage: {{ReasonPhrase}} ResponseCode: {{StatusCode}} RequestId: {{RequestId}} Sha256: {{FileSHA}} FileSize(Bs): {{FileSize}} FileModifiedDate: {{ModifiedDate}}",
    "FailedMessageTemplate": "Azure Storage Logging: Storing blob failed. Reason: ErrorMessageEqualOrGreaterTo1MB ResponseMessage: {{ReasonPhrase}} ResponseCode: {{StatusCode}} RequestId: {{RequestId}}"
  }
```

> [!NOTE] The application (or user if running in Visual Studio) will require `Storage Blob Data Contributor` role on the Event hub. 

Reviewer 3: ```suggestion
> [!NOTE] 
> The Managed Identity principal (or user if running on localhost) will require `Storage Blob Data Contributor` role on the Event hub. 
```
